### PR TITLE
Add documentation about enabling 2fa authentication

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Fixed tickets | fixes #issuenum
| Related PRs | https://github.com/sulu/sulu/pull/6589
| License | MIT

#### What's in this PR?

Add documentation about enabling 2fa authentication.

### CODE REVIEW

**Review by wachterjohannes — APPROVED** [review]

**Comment by wachterjohannes** [comment]
@alexander-schranz you have to rebase your PR and i think @niklasnatter should review it for typos

File: bundles/security/index.rst [hunk 1]
```diff
@@ -106,6 +106,69 @@ the object type and id are also passed the permissions of the security contexts
 from the role might be overridden by the permissions from this specific object
 (which are handled by the previously mentioned ``AccessControlManager``).
 
+
+2FA Authentication
+------------------
+
+Sulu provides 2FA authentication over email via the scheb/2fa packages. They must be
+installed into the project via composer:
```
Comment (niklasnatter) [inline bundles/security/index.rst]:
```suggestion
Two-Factor Authentication
------------------

Sulu allows to use two-factor authentication over email via the scheb/2fa packages. To enable it, 
the packages need to be installed into the project via composer:
```

File: bundles/security/index.rst [hunk 2]
```diff
@@ -106,6 +106,69 @@ the object type and id are also passed the permissions of the security contexts
...
+
+.. code-block:: bash
+
+    composer require scheb/2fa-bundle scheb/2fa-email scheb/2fa-trusted-devices
+
+The security firewall need to be configured to allow 2fa authentication in the
+admin firewall. By enabling following in the ``config/packages//security.yaml``:
```
Comment (niklasnatter) [inline bundles/security/index.rst]:
```suggestion
The security configuration needs to be adjusted to allow two-factor authentication in the
admin firewall. This is configured in the ``config/packages/security.yaml``:
```

File: bundles/security/index.rst [hunk 3]
```diff
@@ -106,6 +106,69 @@ the object type and id are also passed the permissions of the security contexts
...
+   +             two_factor:
+   +                 prepare_on_login: true
+   +                 prepare_on_access_denied: true
+   +                 check_path: 2fa_login_check_admin
+   +                 authentication_required_handler: sulu_security.two_factor_authentication_required_handler
+   +                 success_handler: sulu_security.two_factor_authentication_success_handler
+   +                 failure_handler: sulu_security.two_factor_authentication_failure_handler
+
+The scheb/2fa bundle need to configure email and trusted devices which are the current
+supported 2fa authentication packages ``config/packages/scheb_2fa.yaml``:
```
Comment (niklasnatter) [inline bundles/security/index.rst]:
```suggestion
Afterwards, the scheb/2fa bundle needs to be configured to enable email and trusted devices
in the ``config/packages/scheb_2fa.yaml`` file:
```

File: bundles/security/index.rst [hunk 4]
```diff
@@ -106,6 +106,69 @@ the object type and id are also passed the permissions of the security contexts
...
+.. code-block:: yaml
+
+    scheb_two_factor:
+        email:
+            enabled: true
+            sender_email: "%env(SULU_ADMIN_EMAIL)%"
+        trusted_device:
+            enabled: true
+
+The  ``config/routes/scheb_2fa.yaml``:
```
Comment (niklasnatter) [inline bundles/security/index.rst]:
```suggestion
Additionally, the routes of the scheb/2fa bundle must be added to the project in 
the ``config/routes/scheb_2fa.yaml`` file:
```

File: bundles/security/index.rst [hunk 5]
```diff
@@ -106,6 +106,69 @@ the object type and id are also passed the permissions of the security contexts
...
+
+.. code-block:: yaml
+
+    # For Admin:
+    2fa_login_check_admin:
+        path: /admin/2fa_check
+
+After this configuration and clearing the symfony cache it is possible to enable
+2fa authentication in the profile of the user which is logged in.
```
Comment (niklasnatter) [inline bundles/security/index.rst]:
```suggestion
Finally, after adjusting the configuration and clearing the symfony cache, it is possible to enable
two-factor authentication via the administration interface in the profile of the logged-in user.
```

File: bundles/security/index.rst [hunk 6]
```diff
@@ -106,6 +106,70 @@ the object type and id are also passed the permissions of the security contexts
 from the role might be overridden by the permissions from this specific object
 (which are handled by the previously mentioned ``AccessControlManager``).
 
+
+Two-Factor Authentication
+------------------
```
Comment (alexander-schranz) [inline bundles/security/index.rst]:
```suggestion
Two-Factor Authentication
-------------------------
```

**Pull request merged**